pkg/apis/eksctl.io/v1alpha5/validation.go

```diff
@@ -147,6 +147,23 @@ func ValidateClusterConfig(cfg *ClusterConfig) error {
 		return errors.New("vpc.manageSharedNodeSecurityGroupRules must be enabled when using ekstcl-managed security groups")
 	}
 
+	if err := validateKarpenterConfig(cfg); err != nil {
+		return fmt.Errorf("failed to validate karpenter config: %w", err)
+	}
+
+	return nil
+}
+
+func validateKarpenterConfig(cfg *ClusterConfig) error {
+	if cfg.Karpenter == nil {
+		return nil
+	}
+	if cfg.Karpenter.Version == "" {
+		return errors.New("version field is required if installing Karpenter is enabled")
+	}
+	if len(cfg.FargateProfiles) > 0 && cfg.IAM == nil || (cfg.IAM != nil && IsDisabled(cfg.IAM.WithOIDC)) {
+		return errors.New("OIDC must be set with Karpenter and Fargate Profiles")
+	}
```

Comment:
Does this condition mean that `withOIDC` must be enabled when both `fargateProfiles` and `karpenter` are set?

Reply:
Yes.

Reply:
This check will throw an error if `fargateProfiles` is unset and `withOIDC` is disabled, preventing users from using Karpenter if `withOIDC` is not enabled. For example, this standard Karpenter `ClusterConfig` will fail:

```yaml
karpenter:
  version: <>
iam:
  withOIDC: false # or nil
```

Reply:
This should be:

```suggestion
	if len(cfg.FargateProfiles) > 0 && IsDisabled(cfg.IAM.WithOIDC) {
		return errors.New("iam.withOIDC must be set with Karpenter and Fargate Profiles")
	}
```

The nil check for `cfg.IAM` can be omitted as `SetClusterConfigDefaults` ensures that `ClusterConfig.IAM` is always set.

Reply:
Damn!! Very nice spot Chetan. :) I put the `(` in the wrong place. I meant to put it before the cfg.Iam part. :)

Reply:
Even better :)

Reply:
I was going to suggest adding a unit test for the happy path, but I saw that you just pushed the changes for that.

Reply:
Yep. Thanks. :)